src/components/engine/create-engine-instance.tsx

```diff
@@ -39,12 +45,14 @@ export const CreateEngineInstanceButton = ({
   const trackEvent = useTrack();
   const toast = useToast();
   const accountQuery = useAccount();
+  const [engineTier, setEngineTier] = useState<EngineTier>("STARTER");
 
-  const requestCloudHostedEngine = async () => {
+  const addCloudHostedEngine = async () => {
```

Comment:
make this optionally accept tier
```suggestion
  const addCloudHostedEngine = async (tier?: EngineTier) => {
```